metaflow/plugins/kfp/kfp.py

```diff
@@ -1336,7 +1336,7 @@ def _create_exit_handler_op(self, package_commands: str) -> ContainerOp:
                 f" --flow_name {self.name}"
                 f" --kfp_run_id {dsl.RUN_ID_PLACEHOLDER}"
                 f" --notify_variables_json {json.dumps(json.dumps(notify_variables))}"
-                "  --sqs_message_json {{workflow.parameters.flow_parameters_json}}"
+                "  --sqs_message_json '{{workflow.parameters.flow_parameters_json}}'"
```

Comment:
Come to think of it, `sqs_message_json` is confusing... Because it is not solely the eventing message, but represents the flow parameters.

Because it encapsulates the flow parameters, how about we name it `flow_parameters_json` to be consistent like the rest of the codebase, like here 

* https://github.com/zillow/metaflow/blob/5a78cee4f6a3a5359049fe19ca7eee9ead1ed0c1/metaflow/plugins/kfp/kfp_metaflow_step.py#L230
* https://github.com/zillow/metaflow/blob/c22bf68b8d0f724f6dddd1226e58061a4271345e/metaflow/plugins/kfp/kfp_s3_sensor.py#L140

Reply:
It can be upto the exit handler to include the `flow_parameters_json` in the SQS DLQ, but it may include more (example: the `argo_ui_url` and metaflow `run_id`

Reply:
`sqs_message_json` is the purpose for exit-handler(). Its contents will be sent to sqs. In this case its contents happen to be `flow_parameters_json`, in future, its content could be something else or more flexible. So the parameter name for exit-handler() should be `sqs_message_json` instead of its content `flow_parameters_json`.

Make sense?

Reply:
It makes sense but it took me a while to grok that because it departs from the convention in kfp_metaflow_step.py and kfp_s3_sensor.py.  
So for kfp_exit_handler to be consistent it should use the same parameter name of `flow_parameters_json`

Also, `sqs_message_json` should be constructed within kfp_exit_handler, which may just for now be the whole `flow_parameters_json` but in the future could be more such as include the Metaflow run_id, Argo run, etc.

Reply:
yes, this is exactly my point. To kfp_exit_handler, this parameter represents the value to be sent to sqs. For now its value happens to be the content of `flow_parameters_json`. In future it may be different, the parameter name should be flexible and reflects what it is for.

In other words, for kfp_exit_handler, `sqs_message_json` is the name or key, and the content of `flow_parameters_json` is the value. But for workflow, `flow_parameters_json` is the name, and its value could be anything, i.e. `{"rooms": 3, "zip": "98001"}` etc.

Reply:
The issue is that this breaks with the convention of the codebase making it difficult to read and understand.

* The content of the message to DLQ would be better constructed within the the kfp_exit_handler to also follow the convention of the email message constructed within the kfp_exit_handler.
* `sqs_message_json` when I first read this had me think it is the input message but is truly the `dlq_ sqs_message_json`.

Following the code convention is more maintainable and easier to read and modify, even if one finds a better pattern or  convention (unless the whole code base is refactored to the new convention).

Reply:
> _sqs_message_json when I first read this had me think it is the input message but is truly the dlq_ sqs_message_json._
Your first read was correct, it is the input message for `exit-handler`, and its value is the content of `flow_parameters_json`.

I agree with you that `Following the code convention is more maintainable and easier to read and modify`. The pattern is that the parameter name of a function means what it is for this function.

Reply:
I got why you are confused now. You want to pass `flow_parameters_json` to exit-handler, and let exit-handler decide what content to be written to sqs. This is fine.

What I did now is that the content of sqs is decided in kfp.py, so I name it `sqs_message_json`.

Since now users cannot modify kfp.py or kfp_exit_handler.py any way, and the sqs message is exactly same as `flow_parameters_json`, I do not think which way matters.

What I really want is to let users define their own exit_handler function, then the workflow exit_handler will call users's function.

Reply:
If it helps, I can add more comments to avoid confusion.